### DISCUSSION TITLE
关于环境和编译

### DISCUSSION THREAD

**Discussion by masterchen** [CCXXXI/ecg_monitor#337] [Q&A]
你好，搭建了半天环境 😄,  请问可以写一个详细的运行起来的指南作为新手的指导吗？

安装了flutter \IDEA  \Android Stuio ,IDAE 上边配置好了DART， flutter packages get
出现错误提示说Itnl版本是0.17.0  ，我给改成我当前环境的0.18.0 了， 然后get 没有报错。 最后跑下边的测试，出现错误， 请问是哪儿的问题
D:\OpenSource\github\Flutter\ecg_monitor\test\main_test.dart
/C:/Users/Administrator/AppData/Local/Pub/Cache/git/flutter_ume_kits-f44eeb3231cc36306949729ff27b04694a9415c7/packages/flutter_ume_kit_slow_animation/lib/slow_animation/slow_animation.dart:56:20: Warning: Operand of null-aware operation '?.' has type 'WidgetsBinding' which excludes null.
 - 'WidgetsBinding' is from 'package:flutter/src/widgets/binding.dart' ('/C:/src/flutter/packages/flutter/lib/src/widgets/binding.dart').
    WidgetsBinding.instance
                   ^
/C:/Users/Administrator/AppData/Local/Pub/Cache/git/flutter_ume-7d1801c5e18e44ea4e7833f3f4a5923378827f9d/kits/flutter_ume_kit_ui/lib/third_party/cyclop/lib/src/widgets/color_button.dart:115:13: Warning: Operand of null-aware operation '?.' has type 'OverlayState' which excludes null.
 - 'OverlayState' is from 'package:flutter/src/widgets/overlay.dart' ('/C:/src/flutter/packages/flutter/lib/src/widgets/overlay.dart').
    Overlay.of(rootContext)?.insert(pickerOverlay!);
            ^
/C:/Users/Administrator/AppData/Local/Pub/Cache/git/flutter_ume-7d1801c5e18e44ea4e7833f3f4a5923378827f9d/kits/flutter_ume_kit_ui/lib/third_party/cyclop/lib/src/widgets/eyedrop/eye_dropper_layer.dart:123:13: Warning: Operand of null-aware operation '?.' has type 'OverlayState' which excludes null.
 - 'OverlayState' is from 'package:flutter/src/widgets/overlay.dart' ('/C:/src/flutter/packages/flutter/lib/src/widgets/overlay.dart').
    Overlay.of(context)?.insert(data.eyeOverlayEntry!);
            ^
/C:/src/flutter/packages/flutter_test/lib/src/_goldens_io.dart:8:8: Error: Dart library 'dart:ui' is not available on this platform.
import 'dart:ui';
       ^
Context: The unavailable library 'dart:ui' is imported through these packages:

    main_test.dart => package:ecg_monitor => package:flutter => dart:ui
    ...

Detailed import paths for (some of) the these imports:

    main_test.dart => package:ecg_monitor/generated/l10n.dart => package:flutter/material.dart => package:flutter/src/material/about.dart => package:flutter/foundation.dart => package:flutter/src/foundation/assertions.dart => package:flutter/src/foundation/diagnostics.dart => package:flutter/src/foundation/debug.dart => dart:ui
    main_test.dart => package:ecg_monitor/generated/l10n.dart => package:flutter/material.dart => package:flutter/src/material/about.dart => package:flutter/foundation.dart => package:flutter/src/foundation/binding.dart => dart:ui
    main_test.dart => package:ecg_monitor/generated/l10n.dart => package:flutter/material.dart => package:flutter/src/material/about.dart => package:flutter/foundation.dart => package:flutter/src/foundation/binding.dart => package:flutter/src/foundation/assertions.dart => package:flutter/src/foundation/diagnostics.dart => package:flutter/src/foundation/debug.dart => dart:ui
    main_test.dart => package:ecg_monitor/generated/l10n.dart => package:flutter/material.dart => package:flutter/src/material/about.dart => package:flutter/foundation.dart => package:flutter/src/foundation/binding.dart => package:flutter/src/foundation/debug.dart => dart:ui
    main_test.dart => package:ecg_monitor/generated/l10n.dart => package:flutter/material.dart => package:flutter/src/material/about.dart => package:flutter/foundation.dart => package:flutter/src/foundation/change_notifier.dart => dart:ui
    main_test.dart => package:ecg_monitor/generated/l10n.dart => package:flutter/material.dart => package:flutter/src/material/about.dart => package:flutter/foundation.dart => package:flutter/src/foundation/change_notifier.dart => package:flutter/src/foundation/assertions.dart => package:flutter/src/foundation/diagnostics.dart => package:flutter/src/foundation/debug.dart => dart:ui
    main_test.dart => package:ecg_monitor/generated/l10n.dart => package:flutter/material.dart => package:flutter/src/material/about.dart => package:flutter/foundation.dart => package:flutter/src/foundation/change_notifier.dart => package:flutter/src/foundation/diagnostics.dart => package:flutter/src/foundation/debug.dart => dart:ui
    main_test.dart => package:ecg_monitor/generated/l10n.dart => package:flutter/material.dart => package:flutter/src/material/about.dart => package:flutter/foundation.dart => package:flutter/src/foundation/change_notifier.dart => package:flutter/src/foundation/memory_allocations.dart => dart:ui
    main_test.dart => package:ecg_monitor/generated/l10n.dart => package:flutter/material.dart => package:flutter/src/material/about.dart => package:flutter/foundation.dart => package:flutter/src/foundation/change_notifier.dart => package:flutter/src/foundation/memory_allocations.dart => package:flutter/src/foundation/assertions.dart => package:flutter/src/foundation/diagnostics.dart => package:flutter/src/foundation/debug.dart => dart:ui
    main_test.dart => package:ecg_monitor/generated/l10n.dart => package:flutter/material.dart => package:flutter/src/material/about.dart => package:flutter/foundation.dart => package:flutter/src/foundation/change_notifier.dart => package:flutter/src/foundation/memory_allocations.dart => package:flutter/src/foundation/diagnostics.dart => package:flutter/src/foundation/debug.dart => dart:ui
    main_test.dart => package:ecg_monitor/generated/l10n.dart => package:flutter/material.dart => package:flutter/src/material/about.dart => package:flutter/foundation.dart => package:flutter/src/foundation/debug.dart => dart:ui
    main_test.dart => package:ecg_monitor/generated/l10n.dart => package:flutter/material.dart => package:flutter/src/material/about.dart => package:flutter/foundation.dart => package:flutter/src/foundation/diagnostics.dart => package:flutter/src/foundation/debug.dart => dart:ui
    main_test.dart => package:ecg_monitor/generated/l10n.dart => package:flutter/material.dart => package:flutter/src/material/about.dart => package:flutter/foundation.dart => package:flutter/src/foundation/key.dart => package:flutter/src/foundation/diagnostics.dart => package:flutter/src/foundation/debug.dart => dart:ui
    main_test.dart => package:ecg_monitor/generated/l10n.dart => package:flutter/material.dart => package:flutter/src/material/about.dart => package:flutter/foundation.dart => package:flutter/src/foundation/memory_allocations.dart => dart:ui
    main_test.dart => package:ecg_monitor/generated/l10n.dart => package:flutter/material.dart => package:flutter/src/material/about.dart => package:flutter/foundation.dart => package:flutter/src/foundation/memory_allocations.dart => package:flutter/src/foundation/assertions.dart => package:flutter/src/foundation/diagnostics.dart => package:flutter/src/foundation/debug.dart => dart:ui
    main_test.dart => package:ecg_monitor/generated/l10n.dart => package:flutter/material.dart => package:flutter/src/material/about.dart => package:flutter/foundation.dart => package:flutter/src/foundation/memory_allocations.dart => package:flutter/src/foundation/diagnostics.dart => package:flutter/src/foundation/debug.dart => dart:ui
    main_test.dart => package:ecg_monitor/generated/l10n.dart => package:flutter/material.dart => package:flutter/src/material/about.dart => package:flutter/scheduler.dart => package:flutter/src/scheduler/binding.dart => dart:ui
    main_test.dart => package:ecg_monitor/generated/l10n.dart => package:flutter/material.dart => package:flutter/src/material/about.dart => package:flutter/scheduler.dart => package:flutter/src/scheduler/binding.dart => package:flutter/src/scheduler/priority.dart => package:flutter/foundation.dart => package:flutter/src/foundation/assertions.dart => package:flutter/src/foundation/diagnostics.dart => package:flutter/src/foundation/debug.dart => dart:ui
    main_test.dart => package:ecg_monitor/generated/l10n.dart => package:flutter/material.dart => package:flutter/src/material/about.dart => package:flutter/scheduler.dart => package:flutter/src/scheduler/binding.dart => package:flutter/src/scheduler/priority.dart => package:flutter/foundation.dart => package:flutter/src/foundation/binding.dart => dart:ui
    main_test.dart => package:ecg_monitor/generated/l10n.dart => package:flutter/material.dart => package:flutter/src/material/about.dart => package:flutter/scheduler.dart => package:flutter/src/scheduler/binding.dart => package:flutter/src/scheduler/priority.dart => package:flutter/foundation.dart => package:flutter/src/foundation/binding.dart => package:flutter/src/foundation/assertions.dart => package:flutter/src/foundation/diagnostics.dart => package:flutter/src/foundation/debug.dart => dart:ui
    main_test.dart => package:ecg_monitor/generated/l10n.dart => package:flutter/material.dart => package:flutter/src/material/about.dart => package:flutter/scheduler.dart => package:flutter/src/scheduler/binding.dart => package:flutter/src/scheduler/priority.dart => package:flutter/foundation.dart => package:flutter/src/foundation/binding.dart => package:flutter/src/foundation/debug.dart => dart:ui
    ...

/C:/src/flutter/packages/flutter_test/lib/src/_matchers_io.dart:6:8: Error: Dart library 'dart:ui' is not available on this platform.
import 'dart:ui' as ui;
       ^
/C:/src/flutter/packages/flutter_test/lib/src/accessibility.dart:6:8: Error: Dart library 'dart:ui' is not available on this platform.
import 'dart:ui' as ui;
       ^
/C:/src/flutter/packages/flutter_test/lib/src/accessibility.dart:7:8: Error: Dart library 'dart:ui' is not available on this platform.
import 'dart:ui';
       ^
/C:/src/flutter/packages/flutter_test/lib/src/animation_sheet.dart:6:8: Error: Dart library 'dart:ui' is not available on this platform.
import 'dart:ui' as ui;
       ^
/C:/src/flutter/packages/flutter_test/lib/src/binding.dart:6:8: Error: Dart library 'dart:ui' is not available on this platform.
import 'dart:ui' as ui;
       ^
/C:/src/flutter/packages/flutter_test/lib/src/event_simulation.dart:7:8: Error: Dart library 'dart:ui' is not available on this platform.
import 'dart:ui' as ui;
       ^
/C:/src/flutter/packages/flutter_test/lib/src/frame_timing_summarizer.dart:5:8: Error: Dart library 'dart:ui' is not available on this platform.
import 'dart:ui';
       ^
/C:/src/flutter/packages/flutter_test/lib/src/goldens.dart:6:8: Error: Dart library 'dart:ui' is not available on this platform.
import 'dart:ui';
       ^

**Comment by CCXXXI** [comment]
指南就不写了，所有 Flutter 项目都差不多，也没什么特别值得写的

---

`intl` 的问题之后会处理
- #336

---

报错说 `dart:ui` 不可用，这应该是在 Web 端运行才会出现的情况

**Reply by masterchen** [reply]
好像我导入另外一个项目也会报同样的错误，不是web项目， 可能什么地方配置不对， 第一次搞flutter 😄

**Reply by CCXXXI** [reply]
你是按官方文档配置的环境吗

**Reply by masterchen** [reply]
运行错项目了， “使用Dart Command Line App去运行了”  导致找不到dart:ui 

 但是在运行到模拟器的时候又报了另外一个错误，看起来是又重复引用， 是不是还是什么版本不对呢：
我的环境是：

[√] Flutter (Channel stable, 3.10.2, on Microsoft Windows [版本 10.0.19045.2965], locale zh-CN)

Dart SDK version: 3.0.2 (stable) (Tue May 23 08:26:58 2023 +0000) on "windows_x64"



Launching lib\main.dart on sdk gphone x86 64 in debug mode...
Running Gradle task 'assembleDebug'...
/C:/Users/Administrator/AppData/Local/Pub/Cache/git/flutter_ume-7d1801c5e18e44ea4e7833f3f4a5923378827f9d/kits/flutter_ume_kit_ui/lib/components/widget_detail_inspector/widget_detail_inspector.dart:202:24: Error: 'SearchBar' is imported from both 'package:flutter/src/material/search_anchor.dart' and 'package:flutter_ume_kit_ui/components/widget_detail_inspector/search_bar.dart'.
                child: SearchBar(
                       ^^^^^^^^^
Target kernel_snapshot failed: Exception


FAILURE: Build failed with an exception.

* Where:
Script 'C:\src\flutter\packages\flutter_tools\gradle\flutter.gradle' line: 1201

* What went wrong:
Execution failed for task ':app:compileFlutterBuildDebug'.
> Process 'command 'C:\src\flutter\bin\flutter.bat'' finished with non-zero exit value 1

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 53s
Exception: Gradle task assembleDebug failed with exit code 1

**Reply by CCXXXI** [reply]
拉一下最新的代码
flutter_ume_kit_ui 这个包和新版 Flutter 不兼容，这包应该是字节跳动在维护，但他们好像不怎么管了，我自己稍微修了下
